src/templating/magic_namespace.rb

```diff
@@ -14,15 +14,52 @@
 module Metalware
   module Templating
     class MagicNamespace
-      def initialize(config:, node: nil, firstboot: nil, files: nil)
+      # Every time a new property is added to this namespace it should be added
+      # to this array (so it shows in the `view-config` output), as well as to
+      # the docs in `docs/templating-system.md`.
+      FIELDS = [
```

Comment:
Is their a more implicit way to do this? What is the criteria to be included? Could we make it all public methods that are not part of `Object.methods`?